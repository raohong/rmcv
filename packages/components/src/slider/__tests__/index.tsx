import { render, screen } from '@testing-library/react';
import { Slider, sliderClassNames } from '..';

const testId = 'slider';

it('render correctly', () => {
  const tree = render(<Slider />);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with barHeight', () => {
  const com = render(<Slider barHeight={12} data-testid={testId} />);

  expect(screen.getByTestId(testId)).toHaveStyleRule('height', '12px');

  com.rerender(<Slider barHeight={12} data-testid={testId} vertical />);

  expect(screen.getByTestId(testId)).toHaveStyleRule('width', '12px');
});

it('render with buttonSize', () => {
  render(<Slider buttonSize={20} data-testid={testId} />);

  const content = screen
    .getByTestId(testId)
    .querySelector(`.${sliderClassNames.button}`);

  expect(content).toHaveStyleRule('width', '20px');
  expect(content).toHaveStyleRule('height', '20px');
});

it('render with button', () => {
  render(
    <Slider defaultValue={20} button={v => <span>{v}</span>} data-testid={testId} />,
  );

  expect(screen.getByText('20')).toBeInTheDocument();
});

it('render with leftButton', () => {
  render(
    <Slider
      defaultValue={[10, 20]}
      leftButton={v => <span>{v}</span>}
      data-testid={testId}
      range
    />,
  );

  expect(screen.getByText('10')).toBeInTheDocument();
});

it('render with readonly', () => {
  render(<Slider data-testid={testId} readonly />);

  expect(screen.getByTestId(testId)).toHaveClass(sliderClassNames.readonly);
});

it('render with disabled', () => {
  render(<Slider data-testid={testId} disabled />);

  expect(screen.getByTestId(testId)).toHaveClass(sliderClassNames.disabled);
});
