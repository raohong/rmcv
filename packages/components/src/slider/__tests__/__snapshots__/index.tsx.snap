// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: calc((24px - 2px) / 2) 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  height: 2px;
  border-radius: 999px;
  cursor: pointer;
}

.emotion-1 {
  box-sizing: border-box;
  height: 100%;
  border-radius: inherit;
  background-color: #ebedf0;
}

.emotion-2 {
  box-sizing: border-box;
  position: absolute;
  top: 0;
  height: 100%;
  background-color: #1989fa;
  will-change: left,width,top,height,bottom,height;
  border-radius: inherit;
  transition-property: left,width,top,height,bottom,height;
}

.emotion-3 {
  box-sizing: border-box;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  cursor: -webkit-grab;
  cursor: grab;
  touch-action: pan-y;
  transition-property: left,top;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-4 {
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: unset;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  background-color: #fff;
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.5);
}

<div
    class="RVuiSlider-root emotion-0"
  >
    <div
      class="RVuiSlider-rail emotion-1"
    />
    <div
      class="RVuiSlider-track emotion-2"
      style="left: 0px; width: 0%;"
    />
    <div
      aria-disabled="false"
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="RVuiSlider-thumb emotion-3"
      role="slider"
      style="left: 0%;"
    >
      <button
        class="RVuiSlider-button emotion-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
